### PR TITLE
Migrate appveyor to actions

### PR DESCRIPTION
**Pull request by joseph-flinn**
## Change Summary
Switching over to Github Actions from AppVeyor

## Files Changed
- .github/workflows/build.yml

## Notes
All but `release` based tasks have been tested

Repo Env Vars Needed:
- CHOCO_API_KEY
- NPM_TOKEN
- SNAP_TOKEN

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Set up cloc
+        run: |
+          sudo apt update
+          sudo apt -y install cloc
+
+      - name: Print lines of code
+        run: cloc --include-lang TypeScript,JavaScript,HTML,Sass,CSS --vcs git
```
Comment (kspearrin) [inline .github/workflows/build.yml:25]: I don't think these are the proper params for CLI. According to AppVeyor build, we are using:

```
cloc --include-lang TypeScript,JavaScript --vcs git
```

**Comment by kspearrin** [comment]
On a quick review, looks like we are missing some steps still, around ResourceHacker?

**Comment by joseph-flinn** [comment]
> On a quick review, looks like we are missing some steps still, around ResourceHacker?


Phew, that was a lot of commits. Too bad there isn't a way to do all of the workflow testing locally... I believe that I've covered all of the tasks, but may still be missing something. AppVeyor is difficult to follow with all of the environment dependancies.

File: .github/workflows/build.yml (hunk 1 above)
Comment (joseph-flinn) [reply .github/workflows/build.yml]: Fixed

**Comment by joseph-flinn** [comment]
I forgot one concern that I have. I'm not sure what the `WIN_PKG: C:\Users\appveyor\.pkg-cache\v2.5\fetched-v10.4.1-win-x64` portion of the code is other than a flag for ResourceHacker to run.

**Comment by cscharf** [comment]
> I forgot one concern that I have. I'm not sure what the `WIN_PKG: C:\Users\appveyor\.pkg-cache\v2.5\fetched-v10.4.1-win-x64` portion of the code is other than a flag for ResourceHacker to run.

@kspearrin , any thoughts on this?

**Comment by joseph-flinn** [comment]
> > I forgot one concern that I have. I'm not sure what the `WIN_PKG: C:\Users\appveyor\.pkg-cache\v2.5\fetched-v10.4.1-win-x64` portion of the code is other than a flag for ResourceHacker to run.
> 
> @kspearrin , any thoughts on this?

Actually, I figured it out while working on the the Directory-Connectory.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+      - name: Publish
+        run: |
+          .\scripts\choco-update.ps1 -version $env:PACKAGE_VERSION
+          Get-ChildItem dist
+
+      - name: Publish
+        shell: pwsh
+        run: |
+          # In place of ./scripts/choco-update.ps1
+          cd ./dist
```
Comment (kspearrin) [inline .github/workflows/build.yml:222]: some scripts using forward slash some using back. Just be consistent.

**Review by kspearrin — CHANGES_REQUESTED** [review]
Seem to be missing steps to attaching artifact binaries to releases.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -42,44 +42,44 @@ jobs:
...
       - name: Download & Install RH
         shell: pwsh
         run: |
-          Invoke-WebRequest -Uri http://www.angusj.com/resourcehacker/resource_hacker.zip -OutFile "resource_hacker.zip"
-          Expand-Archive -Path resource_hacker.zip -DestinationPath scripts/resource_hacker
-          echo "D:\a\cli\cli\scripts\resource_hacker" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+          choco install reshack
+          #Invoke-WebRequest -Uri http://www.angusj.com/resourcehacker/resource_hacker.zip -OutFile "resource_hacker.zip"
+          #Expand-Archive -Path resource_hacker.zip -DestinationPath scripts/resource_hacker
+          #echo "D:/a/cli/cli/scripts/resource_hacker" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```
Comment (kspearrin) [inline .github/workflows/build.yml:56]: remove this stuff now?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -226,6 +226,7 @@ jobs:
   # This process seems independent from the others
   build_snap:
     name: Build Snap
+    if: false
```
Comment (kspearrin) [inline .github/workflows/build.yml:229]: ?

**Review by kspearrin — COMMENTED** [review]
looks like you left some stuff from testing still in here.

also, I am still confused about our approach to the pkg cache. do we have to DL it every time? Can we cache some things between runs with github actions? maybe that is a follow up enhancement.

**Pull request closed without merging**